### PR TITLE
owner: fix new owner updating checkpoint too early with pending DDL

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- #1260 was reintroduced into the new owner (released in v5.1.0)

### What is changed and how it works?
- Changed the start-ts of DDL puller to `checkpoint - 1`, to satisfy the open-interval assumption by TiKV.
- Fetch the snapshot of meta data at `checkpoint` instead of `checkpoint-1` avoid overlap between listened DDL events and fetched schema snapshot.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note

- Fix potential DDL loss when owner crashes while executing DDL

### CODE REVIEW

File: tests/kill_owner_with_ddl/run.sh [hunk 1]
```diff
@@ -72,7 +72,7 @@ function run() {
 
     for i in $(seq 1 3); do
         kill_cdc_and_restart $pd_addr $WORK_DIR $CDC_BINARY
-        sleep 2
+        sleep 15
     done
```
Comment (amyangfei) [inline tests/kill_owner_with_ddl/run.sh:76]: this test will sleep `45s` totally, it costs too much time

**Review by amyangfei — APPROVED** [review]

**Comment by ben1009** [comment]
just curious, do we have test cover the case or how do we detect the issue, thanks

**Comment by liuzix** [comment]
> just curious, do we have test cover the case or how do we detect the issue, thanks

kill_owner_with_ddl was supposed to detect this issue, but since it did not sleep long enough for the owner to actually start processing the DDL, the test would pass.

**Review by overvenus — APPROVED** [review]

**Pull request merged**